src/Sulu/Bundle/AdminBundle/Resources/js/components/Tabs/Tabs.js

```diff
@@ -1,56 +1,398 @@
 // @flow
 import React from 'react';
-import type {ChildrenArray, Element} from 'react';
+import type {ChildrenArray, Element, ElementRef} from 'react';
+import {action, computed, observable} from 'mobx';
+import {observer} from 'mobx-react';
+import classNames from 'classnames';
+import debounce from 'debounce';
+import Popover from '../Popover';
+import Icon from '../Icon';
+import type {Skin} from './types';
 import Tab from './Tab';
+import CollapsedTabList from './CollapsedTabList';
+import CollapsedTab from './CollapsedTab';
 import tabsStyles from './tabs.scss';
 
 type Props = {
     children: ChildrenArray<Element<typeof Tab> | false>,
     onSelect: (tabIndex: number) => void,
     selectedIndex: ?number,
+    skin: Skin,
+    small: boolean,
 };
 
-export default class Tabs extends React.PureComponent<Props> {
+const DEBOUNCE_TIME = 200;
+
+@observer
+class Tabs extends React.Component<Props> {
+    static defaultProps = {
+        skin: 'default',
+        small: false,
+    };
+
+    @observable wrapperWidth: number = 0;
+    @observable tabsWrapperWidth: number = 0;
+    @observable tabsWidth: number = 0;
+
+    @observable tabWidths: Map<number, number> = new Map();
+    @observable tabRefs: Map<number, ?ElementRef<'li'>> = new Map();
+    @observable dropdownOpen = false;
+    @observable lastSelectedIndex: ?number;
```

Comment:
Okay, following example: There are 5 tabs shown (1-5) and 5 tabs collapsed (6-10). Click on tab 7 e.g. ... Then the following tabs are shown (1,2,3,4,7) and the others will be collapsed (5,6,8,9,10) ... Then click on tab 3 e.g. ... The visible tabs are still (1,2,3,4,7) and the collapsed tabs are still (5,6,8,9,10), because we don't want the tab 7 to jump back into the collapsed tabs. so tab 3 is the `selectedTab` now and tab 7 is the `lastSelectedIndex` ... `lastShownIndex` would also be ok for me

Reply:
> The visible tabs are still (1,2,3,4,7) and the collapsed tabs are still (5,6,8,9,10), because we don't want the tab 7 to jump back into the collapsed tabs
What do you mean by that? Shouldn't tab 7 be one of the hidden tabs again after tab 3 is clicked?

Reply:
We decided to follow the functionality of the tabs in the chrome dev tools and this is how it's working there.

Reply:
Okay, that is weird, I didn't expect that :see_no_evil: But it indeed is implemented like that in the dev tools... Maybe we have a short chat with Thomas about that once the rest is finished... At least for me it was not the expected behavior, and it makes our code more complex.